https://github.com/wofr06/lesspipe/commit/428062937e99f5a5da5c5f94fecba5b13557d973.patch

Dropped ChangeLog + README hunks, but not weeded out the tiny MP4 change.

From 428062937e99f5a5da5c5f94fecba5b13557d973 Mon Sep 17 00:00:00 2001
From: wf <wp.friebelgmail.com>
Date: Sun, 8 Aug 2021 21:26:17 +0200
Subject: [PATCH] make all tests in the test suite work

---
 lesspipe.sh    | 6 +++---
 lesspipe.sh.in | 6 +++---
 4 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/lesspipe.sh b/lesspipe.sh
index 0bf3cb9..22c10b2 100755
--- a/lesspipe.sh
+++ b/lesspipe.sh
@@ -54,7 +54,7 @@ filecmd() {
 TMPDIR=${TMPDIR:-/tmp}
 sep=:						# file name separator
 altsep==					# alternate separator character
-if [[ -e "$1" && "$1" = *$sep* || "$1" = *$sep*$altsep* ]]; then
+if [[ -e "$1" && "$1" = *$sep* ]]; then
   sep=$altsep
   xxx="${1%=}"
   set "$xxx"
@@ -773,7 +773,7 @@ isfinal() {
   elif [[ "$PARSEHTML" = yes && "$1" = *HTML$NOL_A_P* ]]; then
     msg "append $sep to filename to view the HTML source"
     parsehtml "$2"
-  elif [[ "$1" = *pdf* ]] && cmd_exist pdftotext; then
+  elif [[ "$1" = *pdf* || "$1" = *PDF* ]] && cmd_exist pdftotext; then
     if [[ "$PARSEHTML" = yes ]]; then
       msg "append $sep to filename to view the PDF source"
       istemp "pdftotext -htmlmeta -nopgbrk -q --" "$2" - | parsehtml -
@@ -952,7 +952,7 @@ isfinal() {
   elif [[ "$1" = *mp3* ]] && cmd_exist id3v2; then
       msg "append $sep to filename to view the raw data"
       istemp "id3v2 --list" "$2"
-  elif [[ "$1" = *image* || "$1" = *mp3* || "$1" = *audio* || "$1" = *video* ]] && cmd_exist mediainfo; then
+  elif [[ "$1" = *image* || "$1" = *mp3* || "$1" = *MP4* || "$1" = *audio* || "$1" = *video* ]] && cmd_exist mediainfo; then
     msg "append $sep to filename to view the raw data"
     mediainfo --Full "$2"
   elif [[ "$1" = *image* || "$1" = *mp3* || "$1" = *audio* || "$1" = *video* ]] && cmd_exist exiftool; then
diff --git a/lesspipe.sh.in b/lesspipe.sh.in
index 8819dad..a099a84 100755
--- a/lesspipe.sh.in
+++ b/lesspipe.sh.in
@@ -56,7 +56,7 @@ TMPDIR=${TMPDIR:-/tmp}
 sep=:						# file name separator
 altsep==					# alternate separator character
 # if the separator is part of the file name then altsep is used
-if [[ -e "$1" && "$1" = *$sep* || "$1" = *$sep*$altsep* ]]; then
+if [[ -e "$1" && "$1" = *$sep* ]]; then
   sep=$altsep
   xxx="${1%=}"
   set "$xxx"
@@ -942,7 +942,7 @@ isfinal() {
     msg "append $sep to filename to view the HTML source"
     parsehtml "$2"
 #ifdef pdftotext
-  elif [[ "$1" = *pdf* ]] && cmd_exist pdftotext; then
+  elif [[ "$1" = *pdf* || "$1" = *PDF* ]] && cmd_exist pdftotext; then
     if [[ "$PARSEHTML" = yes ]]; then
       msg "append $sep to filename to view the PDF source"
       istemp "pdftotext -htmlmeta -nopgbrk -q --" "$2" - | parsehtml -
@@ -1202,7 +1202,7 @@ isfinal() {
     mp3info "$2"
 #endif
 #ifdef mediainfo
-  elif [[ "$1" = *image* || "$1" = *mp3* || "$1" = *audio* || "$1" = *video* ]] && cmd_exist mediainfo; then
+  elif [[ "$1" = *image* || "$1" = *mp3* || "$1" = *MP4* || "$1" = *audio* || "$1" = *video* ]] && cmd_exist mediainfo; then
     msg "append $sep to filename to view the raw data"
     mediainfo --Full "$2"
 #ifdef exiftool
